addons/account/models/res_partner_bank.py

```diff
@@ -32,6 +47,30 @@ def _check_journal_id(self):
             if len(bank.journal_id) > 1:
                 raise ValidationError(_('A bank account can belong to only one journal.'))
 
+    @api.constrains('allow_out_payment')
+    def _check_allow_out_payment(self):
+        """ Block enabling the setting, but it can be set to false without the group. (For example, at creation) """
+        for bank in self:
+            if bank.allow_out_payment:
+                if not self.user_has_groups('account.group_validate_bank_account'):
+                    raise ValidationError(_('You do not have the right to trust or un-trust a bank account.'))
+
+    @api.depends('partner_country', 'sanitized_acc_number', 'allow_out_payment')
+    def _compute_display_account_warning(self):
+        for bank in self:
+            if bank.allow_out_payment or not bank.sanitized_acc_number or bank.acc_type != 'iban':
+                bank.display_iban_warning = False
+                bank.display_wise_warning = False
+                continue
+            bank_country = bank.sanitized_acc_number[:2]
+            bank.display_iban_warning = bank_country != bank.partner_country.code
+            bank_institution_code = bank.sanitized_acc_number[4:7]
+            bank.display_wise_warning = bank_institution_code == '967'
+
+    def _compute_user_has_group_validate_bank_account(self):
+        for bank in self:
+            bank.user_has_group_validate_bank_account = self.user_has_groups('account.group_validate_bank_account')
```

Comment:
```suggestion
        self.user_has_group_validate_bank_account = self.user_has_groups('account.group_validate_bank_account')
```
Not really necessary to change but if you want..